csrc/scheduler/matmul.cpp

```diff
@@ -900,11 +900,14 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   // Make a CTA tile
   // ------------------------------------------------------------------
   mma_utils::canonicalizeMmaTvOrdering(mma_result);
+  int nLocalDims = TensorDomain::noDevices(mma_result->getLeafDomain()).size();
   NVF_ERROR(
-      mma_result->nDims() == 3 || mma_result->nDims() == 4,
+      nLocalDims == 3 || nLocalDims == 4,
       "Currently, we only support B, M, N and K being a single dimension.",
       " More general tensor contraction is not supported yet.");
-  const int num_batch_dims = (int)mma_result->nDims() - 3;
+  const int offset_local_dim = (int)mma_result->nDims() - 3;
+  const bool has_batch_dim = (nLocalDims - 3 == 1);
+  const int batch_dim = (int)mma_result->nDims() - nLocalDims;
```

Comment:
Could we place a comment indicating what are the dimensions at this point? I think we are assuming device dims are outermost, so this would be something like
```c++
  // [ (device dims), (batch dims), M, N, K ]
```
Instead of assuming devices are outermost, perhaps we could reorder to make that so adding the following to `mma_utils::canonicalizeMmaTvOrdering`:
```c++
  std::unordered_map<int, int> old2new;
  int next_device_dim = 0;
  for (int i : c10::irange(tv->nDims()) {
    if (tv->axis(i)->isDeviceDim()) {
      old2new[i] = next_device_dim++;
    }
  }
  tv->reorder(old2new);
}
```
That should ensure that the commented dims actually do match reality regardless of input.

Reply:
Thanks! This is a much better solution!